File: Book Coding/Placement series AC/queue.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

#define n 20


class Queue{
    int* arr;
    int front; 
    int back;

    public:
    Queue(){
        arr = new int[n];
        front = -1;
        back = -1;
    }
    
    void push(int x){
        if (back == n-1){
            cout << "The Queue is empty"<<endl;
        }
        back++;
        arr[back] = x;
        
        if (front == -1){
            front++;
        }
    }

    void pop(){
        if (front == -1 || front > back){
            cout << "No element in queue"<<endl;
            return;
        }
        // front++;
        back--;
    }

    void peek(){
        if (front == -1 || front > back){
            cout << "No element in queue"<<endl;
            return;
        }
        cout << arr[front]<<endl;
    }

    bool isEmpty(){
        if (front == -1 || front > back){
            return true;
        }
        return false;
    }

    void printQueue(){
        if (front == -1 || front > back){
            cout << "No element in queue"<<endl;
            return;
        }
        for (int i=0; i<2*sizeof(arr)/sizeof(arr[0]);i++){
            cout << arr[i]<< " ";
        }
        cout <<endl;
    }



};


int main(){
    Queue stump;
    stump.push(1);
    stump.push(2);
    stump.push(6);
    stump.push(3);
    stump.printQueue();
    cout << stump.isEmpty()<<endl;
    stump.peek();
    stump.pop();
    stump.push(5);
    stump.printQueue();
    return 0;
}

File: Level-3 450 set  Pan-1/queue.py

# Enqueue, Deque, Peek, Isfull, IsEmpty, PrintQueueFront, PrintQueueBack
class Queue:
    def __init__(self, capacity):
        self.arr = []
        self.front = -1
        self.back = -1
        self.capacity = capacity

    ## [a, b, c, d, e, f]
    def enqueue(self, x):
        if self.isFull():
            print("Queue is full while enquing")
        else:
            self.arr.insert(0, x)
            # print(self.arr)
            self.front += 1
            # self.back += 1
            
    def deque(self):
        if self.front == -1 or self.back>=self.front:
            print("The Queue is empty!")
        else:
            print("Popped element is: ", self.arr[self.front])
            self.arr.pop(self.front)
            print(self.arr)
            self.front -= 1
    
    def peek(self):
        if self.front == -1:
            print("The Queue is empty!")
        print("Peeked: ", self.arr[self.front-2])

    def isEmpty(self):
        if self.front == -1:
            print("Yes")
        else:
            print("No")
    
    def isFull(self):
        if self.front >= self.capacity - 1:
            print("The Queue is full!")
            return True
        else:
            # print("No")
            return False
    
    def printQueueBack(self):
        if self.front <= -1:
            print("The Queue is empty!")
        for i in self.arr:
            print(i, end=" ")
        print("")
        # print(self.arr[self.front])

    def printQueueFront(self):
        if self.front <= -1:    
            print("The Queue is empty!")
        for i in range(len(self.arr)-1, -1, -1):
            print(self.arr[i], end=" ")
        print("")
        # print(self.arr[self.back1])



if __name__ == '__main__':
    stamp = Queue(5)        ## Capacity = 5
    stamp.enqueue(1)
    stamp.enqueue(2)
    stamp.enqueue(3)
    stamp.enqueue(4)
    stamp.enqueue(5)
    stamp.enqueue(6)

    stamp.isFull()
    stamp.printQueueBack()
    stamp.printQueueFront()
    stamp.deque()
    stamp.peek()
    stamp.isEmpty()
    stamp.printQueueBack()
    stamp.printQueueFront()
    stamp.deque()
    stamp.deque()
    stamp.deque()
    stamp.isFull()
    stamp.deque()
    stamp.deque()
    stamp.printQueueBack()
    stamp.printQueueFront()
    stamp.deque()

    
